keras_nlp/models/xlm_roberta/xlm_roberta_models.py

```diff
@@ -0,0 +1,228 @@
+    defined in the paper, see, for example, `keras_nlp.models.XLMRobertaBase`.
+
+    Args:
+        vocabulary_size: int. The size of the token vocabulary.
+        num_layers: int. The number of transformer layers.
+        num_heads: int. The number of attention heads for each transformer.
+            The hidden size must be divisible by the number of attention heads.
+        hidden_dim: int. The size of the transformer encoding layer.
+        intermediate_dim: int. The output dimension of the first Dense layer in
+            a two-layer feedforward network for each transformer.
+        dropout: float. Dropout probability for the Transformer encoder.
+        max_sequence_length: int. The maximum sequence length this encoder can
+            consume. The sequence length of the input must be less than
+            `max_sequence_length`.
+        name: string, optional. Name of the model.
+        trainable: boolean, optional. If the model's variables should be
+            trainable.
+
+    Example usage:
+    ```python
+    # Randomly initialized XLM-R model
+    model = keras_nlp.models.XLMRobertaCustom(
+        vocabulary_size=50265,
+        num_layers=12,
+        num_heads=12,
+        hidden_dim=768,
+        intermediate_dim=3072,
+        max_sequence_length=12
+    )
+
+    # Call encoder on the inputs.
+    input_data = {
+        "token_ids": tf.random.uniform(
+            shape=(1, 12), dtype=tf.int64, maxval=50265),
+        "padding_mask": tf.constant(
+            [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)),
+    }
+    output = model(input_data)
+    ```
+    """
+
+    pass
+
+
+MODEL_DOCSTRING = """XLM-RoBERTa "{type}" architecture.
+
+    This network implements a bi-directional Transformer-based encoder as
+    described in
+    ["Unsupervised Cross-lingual Representation Learning at Scale"](https://arxiv.org/abs/1911.02116).
+    It includes the embedding lookups and transformer layers, but does not
+    include the masked language modeling head used during pretraining.
+
+    Args:
+        weights: string, optional. Name of pretrained model to load weights.
+            Should be one of {names}.
+        vocabulary_size: int, optional. The size of the token vocabulary.
+        name: string, optional. Name of the model.
+        trainable: boolean, optional. If the model's variables should be
+            trainable.
+
+    Example usage:
+    ```python
+    # Randomly initialized XLMRoberta{type} encoder
+    model = keras_nlp.models.XLMRoberta{type}(weights=None, vocabulary_size=10000)
+
+    # Call encoder on the inputs.
+    input_data = {{
+        "token_ids": tf.random.uniform(
+            shape=(1, 512), dtype=tf.int64, maxval=model.vocabulary_size),
+        "padding_mask": tf.ones((1, 512)),
+    }}
+    output = model(input_data)
+    ```
+"""
+
+
+def XLMRobertaBase(
+    weights="xlm_roberta_base", vocabulary_size=None, name=None, trainable=True
```

Comment:
I think as part of removing the checkpoints, we should default `weights` to `None`, right? Otherwise these will error out.

We can being back the default when we add the checkpoints.